### DISCUSSION TITLE
deepsleep

### DISCUSSION THREAD

**Discussion by giancarlo64** [micropython/micropython#13318] [ESP32]
Hi to all,
I have en ESP32-C3 with firmare 1.22.

I'm trying to use the deepsleep functionality. I need to wake up the ESP32 every 30s or faster if something happen on one pin. So I programmed a wake-up on ext0 on pin GPIO0 that should be a RTC-GPIO. Waking-up every 30 s is ok but I can't wake up from the external source. 
I also want to detect if the same pin during code execution change level so the same pin is programmed to an interrupt.

Here my code:

<pre><code>
import machine, neopixel
from machine import Pin
from machine import deepsleep
from time import sleep
import esp32

p1_event = False

def p1_8_interrupt(pin):
    global p1_event
    #disable interrupts
    pill_p1_8.irq(handler=None)
    pill_p1_event = True

#Wakeup from RTC GPIO GPIO0:
wake1 = Pin(0, mode = Pin.IN)

#wake up pulling down the signal
esp32.wake_on_ext0(pin = wake1, level = esp32.WAKEUP_ALL_LOW)

pill_p1_8 = Pin(0, Pin.IN)
pill_p1_8.irq(trigger=Pin.IRQ_FALLING, handler=p1_8_interrupt)

sleep(10)

#disable p1_8 interrupt
pill_p1_8.irq(handler=None)

deepsleep(30000)
</code></pre>

Why can't I detect a falling edge on GPIO0?

Regards

Gian Carlo

**Comment by davefes** [comment]
You are not pulling the pin high, unless you have an external pull-up on it, ie 10K to 3V3.  Try setting that GPIO to PULLUP or use an external pull-up resistor.

**Comment by giancarlo64** [comment]
Hi, I forgot to mention that the PIN is connected externally with a pull-up resistor. Usually is 3.3V but when I connect the PIN to GND nothing happend

**Reply by davefes** [reply]
I use DS3231 RTC with ESP32.  The alarm goes low and I found that WAKE_ALL_LOW didn't work.  So, I put a transistor inverter and used WAKE_ANY_HIGH.

I tried to find the explanation but I vaguely recall that WAKE_ALL_LOW didn't behave the same as WAKE_ANY_HIGH.  Maybe, search the old forum.

**Comment by giancarlo64** [comment]
Hi,
 I have modified the code and the HW in order to have a pull down Resistor and have a button that pull up the pin in order to generate a rising signal (from 0V to 3.3V). I have modified the code:

`esp32.wake_on_ext0(pin = wake1, level = esp32.WAKEUP_ANY_HIGH)`

but I still can't wake up the ESP32 from external pin . The pin is the G0 of the C3U board (http://docs.m5stack.com/en/core/stamp_c3u) that should be the GPIO0

I also tried to modify the deepsleep without timer `deepsleep()` but still have no success

Could be this the problem?
https://github.com/micropython/micropython/pull/9583

**Reply by davefes** [reply]
Yikes, a lot more complicated than I thought.  I should have said my experience is only with ESP32-WROOM and WROVER.

 Good luck!

**Comment by pop-aruaru** [comment]
I'm success only V1.21.0 and:
a=Pin(2,Pin.IN,pull=Pin.PULL_UP,hold=True)
b=Pin(4,Pin.IN,pull=Pin.PULL_UP,hold=True)
esp32.wake_on_ulp(True)
machine.deepsleep()
V1.22,V1.23,V1.24 didn't wakeup.

**Comment by meirarmon** [comment]
the ESP32-C3 does not support EXT wakeup. I have an open PR (https://github.com/micropython/micropython/pull/17518) that adds `wake_on_gpio` for the C3, used like this (tested on my board):
```
import machine
import esp32
esp32.wake_on_gpio((machine.Pin(5),machine.Pin(2)), esp32.WAKEUP_ANY_HIGH)
machine.deepsleep()
```
